numpy/lib/_arraysetops_impl.pyi

```diff
@@ -279,6 +280,34 @@ def unique(
     equal_nan: bool = ...,
 ) -> tuple[NDArray[Any], NDArray[intp], NDArray[intp], NDArray[intp]]: ...
 
+@overload
+def unique_all(
+    x: _ArrayLike[_SCT], /
+) -> tuple[NDArray[_SCT], NDArray[intp], NDArray[intp], NDArray[intp]]: ...
```

Comment:
The return these here should be named tuples as well, just as is the case during runtime. It's a bit annoying, but that does mean the named tuples definitions will have to be repeated here in the .pyi file:

``` python
# NOTE: still needs a `from typing import Generic` statement

class UniqueAllResult(NamedTuple, Generic[_SCT]):
    values: NDArray[_SCT]
    indices: NDArray[intp]
    inverse_indices: NDArray[intp]
    counts: NDArray[intp]

class UniqueCountsResult(NamedTuple, Generic[_SCT]):
    values: NDArray[_SCT]
    counts: NDArray[intp]

class UniqueInverseResult(NamedTuple, Generic[_SCT]):
    values: NDArray[_SCT]
    inverse_indices: NDArray[intp]
```

Reply:
Sure! I added these classes to typing.